types/shares.go

```diff
@@ -52,13 +52,14 @@ func (m Message) MarshalDelimited() ([]byte, error) {
 	lenBuf := make([]byte, binary.MaxVarintLen64)
 	length := uint64(len(m.Data))
 	n := binary.PutUvarint(lenBuf, length)
-
 	return append(lenBuf[:n], m.Data...), nil
 }
 
 // appendToShares appends raw data as shares.
 // Used for messages.
-func appendToShares(shares []NamespacedShare, nid namespace.ID, rawData []byte) []NamespacedShare {
+func appendToShares(shares []NamespacedShare, id namespace.ID, rawData []byte) []NamespacedShare {
+	nid := make([]byte, len(id))
+	copy(nid, id)
```

Comment:
had to add an extra copy here to stop another overwrite when appending bug. This one was super sneaky, as it only pops up when calling `ComputeShares` more than once. I went a head and added two more copies elsewhere, as they would become problems eventually.

Reply:
IMO, we should use append correctly instead of copy. I think this is easier to grasp (also here it removes the need to introduce another variable for nid):
```go
rawShare := append(append(
	make([]byte, len(nid)+len(rawData)),
	nid...),
	rawData...)
```

Reply:
That certainly makes it more readable, but I don't think it will work for our purposes https://play.golang.org/p/NN4V2gaLh8k

We can however use a new utility function `appendCopy` https://github.com/lazyledger/lazyledger-core/pull/235/commits/6baddea41ae240057b7150fbaaef7c2a3435c13b

Reply:
That's my bad. There is a `0` missing in my suggested code. This is how it should look like: https://play.golang.org/p/oorWoD02jZg

Reply:
ahh okay, I should have guessed, makes sense!